packages/core/src/common/command.ts

```diff
@@ -269,10 +287,12 @@ export class CommandRegistry implements CommandService {
      */
     // tslint:disable-next-line:no-any
     async executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
-        await this.fireWillExecuteCommand(commandId);
+        const executionId = `${commandId}--${this.executionId++}`;
```

Comment:
This line may be a difficult `sell` for PR in Theia.

Reply:
@arekzaluski we discussed this with @westbury , they already do something similar elsewhere in Theia (here we added the `commandId` part mostly for debugging purposes.)

Reply:
Perhaps a colon instead of two dashes would match Theia's conventions better.  See for example https://github.com/eclipse-theia/theia/blob/9404b2e4ddb32fddcf26f9a7da3355aae7300c80/packages/navigator/src/browser/navigator-tree.ts#L50